packages/ckeditor5-list/src/liststyleediting.js

```diff
@@ -509,6 +506,10 @@ function fixListStyleAttributeOnListItemElements( editor ) {
 			return false;
 		}
 
+		if ( baseItem.getAttribute( 'listType' ) !== itemToChange.getAttribute( 'listType' ) ) {
+			return false;
+		}
+
```

Comment:
I'm thinking aloud here:

*   I like that it has the same pattern as above check.
*   I don't like the fact that the last check _could_ be a return.

```javascript
return baseItem.getAttribute( 'listType' ) === itemToChange.getAttribute( 'listType' );
```

However, in this scenario both are OK so it's just a matter of a preference here I think.

For now, this function makes all the checks as I could think of and maybe keeping a same pattern has more benefits here rather than going with preferred `if-else` return form.